# Vaginal_RNASeq_analysis
In this repository we are documenting the analysis of vaginal_rnaseq analysis of tissue cultures exposed to _G. vaginalis_

## Analysis Pipeline

### Raw Reads
Raw reads were generated by "" after RNA extractions by wonderful postdoc Kenetta Nunn (former PhD student of Dr. Larry Forney with both of whom I am collaborating in this project). All files are named as Sample#\_R1.fastq.gz and Sample#\_R2.fastqc.gz, where # correspond to the set {1,2,3,4,6}

### Sequence quality analysis and Trimming

Sequence quality was analyzed with Fastqc. Results were inspected visually and decisions about trimming were performed after visual inspection

`for((i=1;i<=6;i++)); do
  /path_to_fastqc/FastQC/fastqc --outdir ./fastqc --noextract --nogroup Sample''$i''_R1.fastq.gz ; path_to_fastqc/FastQC/fastqc --outdir ./fastqc --noextract --nogroup Sample''$i''_R2.fastq.gz ; 
done`

Trimming was performed after inspecting fastqc reports. a quality threshold of 28 was used to adaptively trim the reads on the 3' end and forward reads were hard trimmed, eliminating the first 9 bp of the sequence at the 5' and the Reverse reads were hard trimmed of the first 14 bp on the 5' end. 

`myTrim="path_to_trim_galore/TrimGalore/trim_galore"
  
for((i=1;i<=6;i++)); do
         $myTrim --paired --quality 28 --phred33 --fastqc \ 
         --fastqc_args "--nogroup --noextract --outdir ./Sample''$i''_trimmed" \
         --adapter CTGTCTCTTATATACACATCT --adapter2 CTGTCTCTTATATACACATCT --stringency 6 --paired --length 60 \
         --clip_R1 10 --clip_R2 15 Sample''$i''_R1.fastq.gz Sample''$i''_R2.fastq.gz \
         --output_dir trimmed/ Sample''$i''_R1.trimmed.fastq Sample''$i''_R2.trimmed.fastq
done`


### Mapping with HISAT2

Reference indexes for the human genome build GRCh38 were obtained from the [hisat2 download](http://daehwankimlab.github.io/hisat2/download/#h-sapiens). More specifically we used the [index built with transcript and snp information](https://genome-idx.s3.amazonaws.com/hisat/grch38_snptran.tar.gz)

Then we mapped trimmed reads to the indexed reference genome

`for i in `cat list`; do
        /patho_to_hisat2/hisat2-2.1.0/hisat2 --phred33 -k 10 --met-file stats/$i.stats --rg-id $i --rg SM:$i --rg PL:illumina \
        -p 1 --rna-strandness RF --fr --dta --un-conc-gz unmapped/$i.unmapped \
        -x /path_to_index/genome_snp_tran -1 trimmed/''$i''_R1_val_1.fq.gz -2 trimmed/''$i''_R2_val_2.fq.gz -S ./mapped_reads/$i.sam
done`

### Post-processing mapped reads

